sklearn/model_selection/tests/test_validation.py

```diff
@@ -713,6 +738,25 @@ def test_learning_curve_with_boolean_indices():
                               np.linspace(0.1, 1.0, 10))
 
 
+def test_learning_curve_with_shuffle():
+    X = np.array([[1, 2], [3, 4], [5, 6], [7, 8], [11, 12], [13, 14], [15, 16],
+                 [17, 18], [19, 20], [7, 8], [9, 10], [11, 12], [13, 14],
+                 [15, 16], [17, 18]])
+    y = np.array([1, 1, 1, 2, 3, 4, 1, 1, 2, 3, 4, 1, 2, 3, 4])
+    groups = np.array([1, 1, 1, 1, 1, 1, 3, 3, 3, 3, 3, 4, 4, 4, 4])
+    estimator = LogisticRegression(multi_class='multinomial', penalty='l2',
```

Comment:
If this also fails at master with an estimator supporting `partial_fit`, such as `SGDClassifier`, then this test can incorporate tests for incremental and non-incremental approaches. I would rather see both tested.